### PR TITLE
Issue #42 - utility method to drop file in file browser

### PR DESCRIPTION
**Pull request by mgibbs1259**
## Pull Request for Issue #42 

**Description of Changes**
Created a method in testutils that drags and drops a file 
Created a test that drags and drops a file into Code, Input Data, and Output Data in a project
Created a test that drags a drops a file into Data in a dataset

**Exit Criteria**
Tested on visible Chrome and Firefox, does not work for headless Chrome

### CODE REVIEW

File: testutils/testutils.py [hunk 1]
```diff
@@ -76,3 +76,92 @@ def is_container_stopped(driver):
...
+
+                Object.setPrototypeOf(d, DragEvent.prototype);
+                f.dispatchEvent(d)
+            });
+            a.parentElement.removeChild(a)
+        };
+        c.documentElement.appendChild(a);
+        a.getBoundingClientRect();
+        return a;"""
+    if project:
```
Comment (billvb) [inline testutils/testutils.py]: I believe this method should somehow wait until the file has definitely dropped successfully - is there some indication in the UI as to what happens?

Comment (mgibbs1259) [reply testutils/testutils.py]: The file drop happens right away, so I don't believe a wait is necessary. In test_file_browser, I assert that the file name is visible in the UI.

File: gigantum_tests/test_file_browser.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+def test_file_drag_drop_project_file_browser(driver: selenium.webdriver, *args, **kwargs):
+    """
+    Test that a file can be dragged and dropped into code, input data,
+    and output data in a project.
+
+    Args:
+        driver
+    """
+    # Project set up
+    testutils.log_in(driver)
```
Comment (billvb) [inline gigantum_tests/test_file_browser.py]: You can now use snippets to get rid of all this boilerplate, eg

```
    r = testutils.prep_py3_minimal_base(driver)
    username, project_title = r.username, r.project_name
```

That will get you to the point where you are on a project page and the base successfully built

File: testutils/elements.py [hunk 3]
```diff
@@ -318,11 +322,18 @@ def close_collaborators_button(self):
     def import_first_cloud_project_button(self):
         return self.driver.find_element_by_css_selector(".RemoteLabbooks__icon--cloud-download")
 
+class CodeElements(UiElement):
```
Comment (billvb) [inline testutils/elements.py]: Can we make these all one class, like ProjectFileBrowserElements or something?

File: testutils/testutils.py [hunk 4]
```diff
@@ -91,3 +94,82 @@ def is_container_stopped(driver):
...
+        a.getBoundingClientRect();
+        return a;"""
+    drop_target = driver.find_element_by_css_selector(".FileBrowser")
+    logging.info("Adding file")
+    file_path = '/tmp/sample-upload.txt'
+    with open(file_path, 'w') as example_file:
+        example_file.write('Sample Text')
+    file_input = driver.execute_script(js_script, drop_target, 0, 0)
+    file_input.send_keys(file_path)
+    wait = WebDriverWait(driver, 200)
```
Comment (billvb) [inline testutils/testutils.py]: can we chagen the wait to something a lot slower, like 10?

File: testutils/elements.py (hunk 3 above)
Comment (mgibbs1259) [reply testutils/elements.py]: Yes

Comment (billvb) [reply testutils/elements.py]: I've had idea, but for another PR

**Pull request merged**